symphony/bdk/core/service/obo_services.py

```diff
@@ -0,0 +1,69 @@
+from symphony.bdk.core.auth.auth_session import OboAuthSession
+from symphony.bdk.core.client.api_client_factory import ApiClientFactory
+from symphony.bdk.core.config.model.bdk_config import BdkConfig
+from symphony.bdk.core.service.connection.connection_service import OboConnectionService
+from symphony.bdk.core.service.message.message_service import OboMessageService
+from symphony.bdk.core.service.stream.stream_service import OboStreamService
+from symphony.bdk.core.service.user.user_service import OboUserService
+from symphony.bdk.core.service_factory import OboServiceFactory
+
+
+class OboServices:
+    """Entry point for OBO-enabled services, see
+    `the list of OBO-enabled endpoints <https://developers.symphony.com/restapi/reference#obo-enabled-endpoints`_
+    """
+
+    async def __aenter__(self):
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self.close_clients()
+
+    def __init__(self, config: BdkConfig, obo_session: OboAuthSession):
+        """
+
+        :param config: the BDK configuration.
+        :param obo_session: the OBO session to use.
+        """
+        self._config = config
+        self._obo_session = obo_session
+
+        self._api_client_factory = ApiClientFactory(config)
```

Comment:
should we recreate one or reuse the one from BDK? (I'm wondering why we did it like that for the Java BDK)

Reply:
In java, we recreate a new one: 

- https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/master/symphony-bdk-core/src/main/java/com/symphony/bdk/core/SymphonyBdk.java#L258
- https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/364fc4309fb44bab2054d310d8deededc6935c88/symphony-bdk-core/src/main/java/com/symphony/bdk/core/OboServices.java#L31

Reply:
Yep that's what I was wondering, do we have a good reason to do it?
If we reuse the existing one from BDK it means we don't have to use async with